### PR TITLE
Add Velocity Controller and Fix I term in PID code

### PR DESCRIPTION
**Pull request by ashaw596**

### CODE REVIEW

**Comment by justbuchanan** [comment]
Were you able to find the article/paper on doing a position and velocity PID on top of each other?  I'm a little sketchy on this double PID thing...

**Comment by PhilipBale** [comment]
You should probably study for 2200

**Comment by justbuchanan** [comment]
Bro, I've got 2200 on lock down :godmode:

File: common/Pid.cpp [hunk 1]
```diff
@@ -35,16 +35,19 @@ Pid::~Pid()
...
 {
-	if (w > 0)
+	if (w != _windup) 
 	{
-		if (w != _windup) {
-			_windup = w;
-			_oldErr = new float[_windup];
-			memset(_oldErr, 0, sizeof(float)*_windup);
+		if (_oldErr) 
+			delete[] _oldErr;
```
Comment (justbuchanan) [inline common/Pid.cpp]: dangling pointer

**Comment by justbuchanan** [comment]
@ashaw596, do you want to refactor this so that it's just the PID fixes?  I like those changes, but don't think the velocity PID thing is of much use, especially since that will be wiped out by the new motion control stuff.

**Pull request closed without merging**